core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java

```diff
@@ -109,13 +128,10 @@ private HttpResponse convertHeaders(ResponseHeaders headers, boolean endStream,
         return res;
     }
 
-    private void convertHeaders(HttpHeaders inHeaders, io.netty.handler.codec.http.HttpHeaders outHeaders,
+    private void convertHeaders(ResponseHeaders inHeaders, io.netty.handler.codec.http.HttpHeaders outHeaders,
                                 boolean isTrailersEmpty) {
-        if (keepAliveHandler.needsDisconnection()) {
-            sentConnectionCloseHeader = true;
-        }
         ArmeriaHttpUtil.toNettyHttp1ServerHeaders(inHeaders, outHeaders, http1HeaderNaming,
-                                                  !sentConnectionCloseHeader);
+                                                  !keepAliveHandler.needsDisconnection());
```

Comment:
Reading this block, I realized that we have a bug. 😓 
This is the patch for the bug. https://github.com/line/armeria/pull/4864